File: python/resistor-color-trio/resistor_color_trio.py
codes = [
    "black",
    "brown",
    "red",
    "orange",
    "yellow",
    "green",
    "blue",
    "violet",
    "grey",
    "white",
]


def label(colors):
    n = (codes.index(colors[0]) * 10 + codes.index(colors[1])) * (10 ** codes.index(colors[2]))
    if n > 1e9:
        return f"{n // 1e9:.0f} gigaohms"
    if n > 1e6:
        return f"{n // 1e6:.0f} megaohms"
    if n > 1e3:
        return f"{n // 1e3:.0f} kiloohms"
    return f"{n} ohms"

File: c/resistor-color-trio/resistor_color_trio.c
#include "resistor_color_trio.h"

#include <math.h>

resistor_value_t color_code(resistor_band_t bands[]) {
  resistor_value_t value = {(bands[0] * 10 + bands[1]) * pow(10, bands[2]),
                            OHMS};
  if (value.value >= 10E8) {
    value.value /= 10E8;
    value.unit = GIGAOHMS;
  } else if (value.value >= 10E5) {
    value.value /= 10E5;
    value.unit = MEGAOHMS;
  } else if (value.value >= 10E2) {
    value.value /= 1000;
    value.unit = KILOOHMS;
  }
  return value;
}
